### PR TITLE
feat(adapters): land planning artifacts for multi-framework adapters

### LINKED ISSUE #260: feat(adapters): multi-framework adapters for Cursor and Codex
## Summary

Add a Node.js generation script (`npm run adapters:sync`) and companion drift check (`npm run adapters:check`) that convert the canonical Specorator Claude baseline (`.claude/agents/`, `.claude/skills/`, `AGENTS.md`) into framework-adapter files for Cursor (`.cursor/rules/*.mdc`) and Codex (new files under `.codex/agents/` and `.codex/skills/`). Generated files are additive — no canonical Claude file is touched. Drift detection wires into `npm run verify` and is CI-provider agnostic.

## Why

Per `docs/specorator-product/product.md` Strategic Priority 3, Specorator should stay portable across AI coding tools while preserving `AGENTS.md` and the workflow docs as source of truth. Today, Cursor and Codex users either ignore `.claude/` surfaces or hand-translate them — and the translations drift silently when canonical files change.

## Planning artifacts

Full lifecycle planning (Stages 1–6) lives under `specs/multi-framework-adapters/`:

- **Idea** — `specs/multi-framework-adapters/idea.md` (IDEA-ADAPT-001)
- **Research** — `specs/multi-framework-adapters/research.md` (RESEARCH-ADAPT-001) — 8 questions answered, Cursor `.mdc` format, Codex `AGENTS.md` walk, `cursor-doctor` validator, sync strategy
- **Requirements** — `specs/multi-framework-adapters/requirements.md` v2.2 (PRD-ADAPT-001) — REQ-ADAPT-001a..017, NFR-ADAPT-001..007, NG1..NG9
- **Design** — `specs/multi-framework-adapters/design.md` (DESIGN-ADAPT-001) — Parts A (UX/CLI flows), B (terminal output, file structure), C (architecture, 12-step pipeline)
- **Specification** — `specs/multi-framework-adapters/spec.md` (SPECDOC-ADAPT-001) — 10 interfaces, 5 data structures, 23 edge cases, 40 test scenarios
- **Tasks** — `specs/multi-framework-adapters/tasks.md` (TASKS-ADAPT-001) — 33 tasks across 10 phases, TDD-ordered

## ADRs

- **ADR-0028** — Treat the Claude baseline as the canonical source for multi-framework adapters
- **ADR-0029** — Make the multi-framework adapter pipeline additive-only over canonical sources

## Scope highlights

- **Cursor output**: `.cursor/rules/agent-<slug>.mdc`, `.cursor/rules/skill-<slug>.mdc`, `.cursor/rules/project-conventions.mdc` (flat layout, agent-requested rules, one always-apply for project conventions)
- **Codex output**: `.codex/agents/<slug>.md`, `.codex/skills/<slug>.md`, `.codex/agents/INDEX.md` (discovered via manual one-time `AGENTS.md` pointer)
- **Manifest**: `.cursor/rules/.adapter-manifest.json` with top-level `script_hash`, `sources[]`, `outputs[]`, `generated_at`
- **Validation**: `cursor-doctor` pinned `devDependency`
- **CI integration**: via `npm run verify` only — no provider-specific files (NG9)

## Out of scope (this issue)

- Aider, Copilot, Gemini, or any framework beyond Cursor/Codex (require separate ADRs)
- Two-way sync (canonical → adapters only)
- GUI tooling
- Modification of any canonical file

## This PR

Ships planning artifacts + ADRs only. Implementation (T-ADAPT-001..033) lands in follow-up PRs per `tasks.md`.

### PR DESCRIPTION
**Pull request by Luis85**
Closes #260.

## Summary

- Lands Stages 1–6 planning artifacts for the **Multi-Framework Adapters** feature.
- Adds a Node.js generation script and drift check spec that converts the canonical Claude baseline (`.claude/agents/`, `.claude/skills/`, `AGENTS.md`) into framework-adapter files for Cursor (`.cursor/rules/*.mdc`) and Codex (`.codex/agents/`, `.codex/skills/`, `INDEX.md`). Generated files are additive; no canonical Claude file is touched.
- Drift detection wires into `npm run verify` and is **CI-provider agnostic** (NG9): no `.github/workflows/`, `Jenkinsfile`, or `.circleci/` files are required or shipped.
- Two ADRs: ADR-0028 (Claude baseline canonical) and ADR-0029 (additive-only pipeline).

## Artifacts (`specs/multi-framework-adapters/`)

| Stage | File | ID |
|---|---|---|
| Idea | `idea.md` | IDEA-ADAPT-001 |
| Research | `research.md` | RESEARCH-ADAPT-001 |
| Requirements | `requirements.md` v2.2 | PRD-ADAPT-001 |
| Design | `design.md` | DESIGN-ADAPT-001 |
| Specification | `spec.md` | SPECDOC-ADAPT-001 |
| Tasks | `tasks.md` | TASKS-ADAPT-001 |

24 functional REQs, 7 NFRs, 9 non-goals, 23 edge cases, 40 test scenarios, 33 tasks across 10 phases (TDD-ordered).

## Out of scope (this PR)

Implementation tasks T-ADAPT-001..033 land in follow-up PRs per `tasks.md`. This PR ships planning artifacts + ADRs only.

## What was done end-to-end

The full Specorator lifecycle (Stages 1–6) ran via `/orchestrate`, with one `/spec:clarify` gate after Requirements and one `/spec:analyze` gate before Tasks. Outcomes:

- Clarify gate raised 29 items (3 blockers); all resolved in PRD v2.
- Analyze gate raised 7 items (3 blockers, 4 nice-to-resolve); all resolved across requirements/design/spec.
- During the verify pass, REQ IDs with letter suffixes (`REQ-ADAPT-001a/b/c` from the PM split convention) were renumbered to sequential 3-digit IDs to match the project's `traceabilityIdPattern`. 121 references updated across 5 files.

## Test plan

- [ ] CI verify gate (Linux) green.
- [ ] `check:traceability` ✓ (verified locally on Windows).
- [ ] `check:links` ✓ (verified locally on Windows).
- [ ] `check:adr-index` ✓ (ADR-0028 and ADR-0029 indexed).
- [ ] `check:frontmatter` ✓.
- [ ] `check:agents` ✓.
- [ ] `check:automation-registry` ✓.
- [ ] Reviewer confirms NG9 (CI-provider agnostic) holds.

## Notes for reviewer

- Local `test:scripts` is flaky on Windows (different unrelated tests fail per run with `STATUS_STACK_BUFFER_OVERRUN` / fork failures); CI on Linux is the real gate for that step.
- Implementation is gated to follow-up PRs; this PR is documentation + ADR only.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: specs/multi-framework-adapters/requirements.md [hunk 1]
```diff
@@ -0,0 +1,578 @@
...
+- NG4. Modification of any existing hand-authored file under `.codex/` (`.codex/README.md`, `.codex/instructions.md`, `.codex/workflows/*.md`). Those files are left as-is.
+- NG5. Modification of any canonical file: `.claude/`, `AGENTS.md`, `CLAUDE.md`, `memory/constitution.md`. The adapter script is additive-only.
+- NG6. Runtime orchestration: the adapters provide configuration context files, not execution wiring between Claude Code agents and Cursor or Codex agents.
+- NG7. Generating Specorator `templates/` artifacts for other frameworks; the adapters cover AI tool configuration files only.
+- NG8. Semantic validation of generated rule content (whether a rule will be effective in practice). Structural validation only.
+- NG9. CI-provider-specific configuration: this feature ships no CI-provider-specific files (e.g., `.github/workflows/`, `Jenkinsfile`, `.circleci/`). CI integration is achieved solely by invoking `npm run verify`, which any CI provider or release pipeline can call.
+- NG9. The adapter script shall not modify `AGENTS.md`; the `AGENTS.md` pointer to `.codex/agents/INDEX.md` is a manual one-time setup step documented in the adapter README, not automated.
```
[the remark this answers is not in the document]
Comment (Luis85) [reply specs/multi-framework-adapters/requirements.md]: Fixed in `348377e`. The duplicate non-goal ID has been split so CI-provider agnosticism remains `NG9` and AGENTS.md immutability is `NG10`; `npm run verify` passes locally.

File: docs/adr/0029-adapter-pipeline-additive-only.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+## Status
+
+Accepted
+
+## Context
+
+The Multi-Framework Adapters feature (PRD-ADAPT-001) introduces a Node.js generation script that converts canonical Claude-baseline files into Cursor and Codex adapter outputs. Two boundary properties of this pipeline are architecturally load-bearing:
+
+1. The script must never modify any canonical source file (`.claude/`, `AGENTS.md`, `CLAUDE.md`, `memory/constitution.md`). Captured in PRD non-goal NG5.
+2. The script must never modify any hand-authored Codex file (`.codex/README.md`, `.codex/instructions.md`, `.codex/workflows/*.md`). Captured in PRD non-goal NG4 and REQ-ADAPT-007b.
```
[the remark this answers is not in the document]
Comment (Luis85) [reply docs/adr/0029-adapter-pipeline-additive-only.md]: Fixed in `348377e`. The ADR now references the active `REQ-ADAPT-026` ID and the current non-goal set; `npm run verify` passes locally.

File: specs/multi-framework-adapters/requirements.md [hunk 3]
```diff
@@ -0,0 +1,578 @@
...
+
+### REQ-ADAPT-005 — Cursor adapter individual file size limit
+
+- **Pattern:** Ubiquitous
+- **Statement:** The adapter generation script shall produce each Cursor `.mdc` file with a total line count (including frontmatter) of no more than 500 lines, and if any source file would produce more than 500 lines, the script shall truncate the output at 490 lines and append `<!-- TRUNCATED: source exceeded 500 lines -->` as line 491.
+- **Acceptance:**
+  - Given all generated `.mdc` files in `.cursor/rules/`
+  - When each file's total line count (including frontmatter) is measured
+  - Then no file exceeds 491 lines
+  - And any file whose source would have produced more than 500 lines contains the text `<!-- TRUNCATED: source exceeded 500 lines -->` on line 491
```
[the remark this answers is not in the document]
Comment (Luis85) [reply specs/multi-framework-adapters/requirements.md]: Fixed in `348377e`. REQ-ADAPT-005 now states the rendered `.mdc` arithmetic explicitly, and the task/state artifacts no longer expect a 491-line rendered Cursor file; `npm run verify` passes locally.

**Comment by Luis85** [comment]
@codex review
